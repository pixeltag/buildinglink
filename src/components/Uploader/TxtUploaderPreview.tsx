import React from 'react'

export default function TxtUploaderPreview({ fileSize, fileName, handleRemoveFile }: { fileSize: number, fileName: string, handleRemoveFile: () => void }) {
    return (
        <ul className="uploader-preview">
            <li>
                <div className="uploader-preview_list">
                    <div className='uploader-preview_wrapper'>
                        <div>
                            <div className="uploader-preview_icon">
                                <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
                                    <path d="M5 4a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm-.5 2.5A.5.5 0 0 1 5 6h6a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5zM5 8a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1H5z" />
                                    <path d="M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1z" />
                                </svg>
                            </div>
                        </div>

                        <div>
                            <p className="uploader-preview_filename">
                                {fileName}
                            </p>
                            <p className="uploader-preview_filesize">
                                {fileSize} bytes
                            </p>
                        </div>
                    </div>
                    <div className="uploader-preview_action">
                        <button
                            type="button"
                            onClick={() => handleRemoveFile()}
                            className="uploader-preview_button">
                            <svg className="w-3 h-3" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 14 14">
                                <path stroke="currentColor" strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6" />
                            </svg>
                        </button>
                    </div>
                </div>
            </li>
        </ul>
    )
}
