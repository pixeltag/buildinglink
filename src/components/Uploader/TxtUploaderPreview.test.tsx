import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import TxtUploaderPreview from "./TxtUploaderPreview";


test('Shoud render txtUploaderPreview compoent', async () => {
    const handleRemoveFileMockFn = jest.fn()
    render(<TxtUploaderPreview
        fileName='name'
        fileSize={10}
        content="test"
        handleRemoveFile={() => handleRemoveFileMockFn} />)
    const fileNameElement = screen.getByTestId("file-name");
    const fileSizeElement = screen.getByTestId("file-size");
    const contentElement = screen.getByTestId("content");

    expect(fileNameElement).toBeInTheDocument();
    expect(fileNameElement).toHaveTextContent('name')

    expect(fileSizeElement).toBeInTheDocument();
    expect(fileSizeElement).toHaveTextContent('10 bytes')

    expect(contentElement).toBeInTheDocument();
    expect(contentElement).toHaveTextContent('test')
})



test('Shoud fire remove fn when click on remove btn', async () => {
    const handleRemoveFileMockFn = jest.fn()
    render(<TxtUploaderPreview
        fileName='name'
        fileSize={10}
        content="test"
        handleRemoveFile={() => handleRemoveFileMockFn()} />)

    const removeBtn = screen.getByTestId('remove-btn');
    fireEvent.click(removeBtn)
    expect(handleRemoveFileMockFn).toHaveBeenCalledTimes(1);
})


test('It renders correctly', () => {
    const handleRemoveFileMockFn = jest.fn()
    const element = render(<TxtUploaderPreview
        fileName='name'
        fileSize={10}
        content="test"
        handleRemoveFile={() => handleRemoveFileMockFn()} />)
    expect(element).toMatchSnapshot();
})