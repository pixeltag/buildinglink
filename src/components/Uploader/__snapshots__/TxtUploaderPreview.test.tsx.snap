// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="uploader-preview"
      >
        <li>
          <div
            class="uploader-preview_list"
          >
            <div
              class="uploader-preview_wrapper"
            >
              <div>
                <div
                  class="uploader-preview_icon"
                >
                  <svg
                    aria-hidden="true"
                    viewBox="0 0 16 16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5 4a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm-.5 2.5A.5.5 0 0 1 5 6h6a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5zM5 8a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1H5z"
                    />
                    <path
                      d="M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1z"
                    />
                  </svg>
                </div>
              </div>
              <div>
                <p
                  class="uploader-preview_filename"
                  data-testid="file-name"
                >
                  name
                </p>
                <p
                  class="uploader-preview_filesize"
                  data-testid="file-size"
                >
                  10
                   bytes
                </p>
              </div>
            </div>
            <div
              class="uploader-preview_action"
            >
              <button
                class="uploader-preview_button"
                data-testid="remove-btn"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="w-3 h-3"
                  fill="none"
                  viewBox="0 0 14 14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="uploader-preview_content"
            data-testid="content"
          >
            test
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="uploader-preview"
    >
      <li>
        <div
          class="uploader-preview_list"
        >
          <div
            class="uploader-preview_wrapper"
          >
            <div>
              <div
                class="uploader-preview_icon"
              >
                <svg
                  aria-hidden="true"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 4a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm-.5 2.5A.5.5 0 0 1 5 6h6a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5zM5 8a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1H5z"
                  />
                  <path
                    d="M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1z"
                  />
                </svg>
              </div>
            </div>
            <div>
              <p
                class="uploader-preview_filename"
                data-testid="file-name"
              >
                name
              </p>
              <p
                class="uploader-preview_filesize"
                data-testid="file-size"
              >
                10
                 bytes
              </p>
            </div>
          </div>
          <div
            class="uploader-preview_action"
          >
            <button
              class="uploader-preview_button"
              data-testid="remove-btn"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="w-3 h-3"
                fill="none"
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="uploader-preview_content"
          data-testid="content"
        >
          test
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
